docs/design/editorial_design.md

```diff
@@ -0,0 +1,354 @@
+
+- Overwrite
+- Trim
+- Cut
+- Slip
+- Slide
+- Ripple
+- Roll
+- Fill (3/4 Point Edit)
+
+OpenTimelineIO, while not seeking to become a fully realized NLE, could benefit from having a simple toolkit to help automate building/modifying timelines based on time rather than index alone.
+
+## Structure
+The underlying implementation for these edit commands will be in C++, possibly under a new library called `openedit`. This has the benefit of connecting with other languages more simply (Python, Swift, C, etc.) and, hopefully, we only need expose the functionality to the bindings.
+
+```
+OpenTimelineIO/
+    `- src/
+        `- opentime/
+        `- opentimelineio/
+        `- openedit/
+            `- ...
+```
+
+The benefit for the extra library is the edit suite will inevitably grow and will begin crowding the core model, which ultimately should be able to run without it.
+
+## Dependencies / Namespaces
+The `openedit` library will have to communicate with both `opentime` and `openedit`.
+
+This will warrant either a new namespace `namespace openedit { ... }` or using `opentimelineio` again. The latter makes some of the bellow comments less challenging but again, might pollute an otherwise clean namespace.
+
+### ErrorStatus
+This will present challenges when interfacing with the `ErrorStatus` of the different namespaces. That said, with a simple wrapper for this, we could have the instances auto convert as required.
+
+### Retainer
+Ideally, we continue to use the `Retainer<>` where possible to make sure editing is reference counted. Namespaces might make the names a bit clunky, seeing `SerializedObject::Retainer<>` all over:
+
+```
+namespace openedit { OPENEDIT_VERSION {
+
+using namespace opentimelineio::OPENTIMELINEIO_VERSION;
+auto item = SerializedObject::Retainer<Item>(...);
+
+} }
+```
+
+So perhaps establishing some common aliases early might help?
+```
+using RetainedItem = SerializedObject::Retainer<Item>;
+using RetainedComposition = SerializedObject::Retainer<Composition>;
+...
+```
+
+# Editing Commands
+Let's go through some of the most common commands and what the python API might look like:
+
+## Overwrite
+
+![Overwrite](../_static/edit/01_overwrite.png)
+- Anything overlapping is destroyed on contact. This may split an item.
+
+#### API
+```py
+"""
+Args:
+    item: Item that we're going to place onto the track
+    track: Track that this item will belong to afterwards
+    track_time: RationalTime of our track to put this item at
+    fill_template: Optional[Item] that will be cloned where required
+        to fill in the event that this overwrite extends the end of the
+        composition's limit
+"""
+otio.algorithms.overwrite(
+    item: Item,
+    track: Track,
+    track_time: RationalTime,
+    fill_template: Item = None, # Default to Gap
+)
+```
+- In the image, Clip `C` is `item` and `track` is the what `A` and `B` are on.
+
+----
+
+## Insert
+![Insert](../_static/edit/02_insert.png)
```

Comment:
The pngs are very nice, but I'm wondering if we could get some inline ASCII art as well :)